docs/notebooks/quickstart.md

```diff
@@ -0,0 +1,206 @@
+---
+jupytext:
+  text_representation:
+    format_name: myst
+kernelspec:
+  display_name: Python 3
+  name: python3
+---
+
+# JAX Quickstart
+
+**JAX is NumPy on the CPU, GPU, and TPU, with great automatic differentiation for high-performance machine learning research.**
+
+With its updated version of [Autograd](https://github.com/hips/autograd), JAX
+can automatically differentiate native Python and NumPy code. It can
+differentiate through a large subset of Python’s features, including loops, ifs,
+recursion, and closures, and it can even take derivatives of derivatives of
+derivatives. It supports reverse-mode as well as forward-mode differentiation, and the two can be composed arbitrarily
+to any order.
+
+What’s new is that JAX uses
+[XLA](https://www.tensorflow.org/xla)
+to compile and run your NumPy code on accelerators, like GPUs and TPUs.
+Compilation happens under the hood by default, with library calls getting
+just-in-time compiled and executed. But JAX even lets you just-in-time compile
+your own Python functions into XLA-optimized kernels using a one-function API.
+Compilation and automatic differentiation can be composed arbitrarily, so you
+can express sophisticated algorithms and get maximal performance without having
+to leave Python.
+
+```{code-cell}
+import jax.numpy as jnp
+from jax import grad, jit, vmap
+from jax import random
+```
+
+```{code-cell}
+:tags: [remove-cell]
+# Execute this to consume & hide the GPU warning.
+jnp.arange(10)
+```
+
+## Multiplying Matrices
+
+We'll be generating random data in the following examples. One big difference between NumPy and JAX is how you generate random numbers. For more details, see [Common Gotchas in JAX].
+
+[Common Gotchas in JAX]: https://jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html#%F0%9F%94%AA-Random-Numbers
+
+```{code-cell}
+key = random.PRNGKey(0)
+x = random.normal(key, (10,))
+print(x)
+```
+
+Let's dive right in and multiply two big matrices.
+
+```{code-cell}
+size = 3000
+x = random.normal(key, (size, size), dtype=jnp.float32)
+%timeit jnp.dot(x, x.T).block_until_ready()  # runs on the GPU
+```
+
+We added that `block_until_ready` because JAX uses asynchronous execution by default (see {ref}`async-dispatch`).
+
+JAX NumPy functions work on regular NumPy arrays. 
+
+```{code-cell}
+import numpy as np
+x = np.random.normal(size=(size, size)).astype(np.float32)
+%timeit jnp.dot(x, x.T).block_until_ready()
+```
+
+That's slower because it has to transfer data to the GPU every time. You can ensure that an NDArray is backed by device memory using {func}`~jax.device_put`.
```

Comment:
The {func}`~jax.device_put` does not seem to produce a link

Reply:
I am judging this by the GitHub view feature, which is why I think that I am not seeing the links

Reply:
I found a [link](https://jax--5537.org.readthedocs.build/en/5537/notebooks/quickstart.html) to the rendered documentation.

Reply:
Yes, all of these files are text source (similar to RST) and cross-references & links will not render except in the sphinx build.